### PR TITLE
irem/travrusa.cpp: Add support for Traverse USA and clones different PROMs setup [Victor Fernandez (City Game), Recreativas.org]

### PR DESCRIPTION
**Pull request by clawgrip**
Add support for Traverse USA and clones different PROMs setup with two 256x4 character palette PROMs instead of only 256x8 one (same contents on both setups).

Confirmed to exist on Traverse USA, MotoTour and MotoRace USA.

### CODE REVIEW

**Comment by cuavas** [comment]
Is it just a case of the organisation being different which could be addressed during ROM loading?  It would be good if someone could give this a look and see if it’s simple to get working before merging.

**Comment by Osso13** [comment]
If you load them like this it will work:
	ROM_REGION( 0x0320, "proms", 0 )
	ROM_LOAD_NIB_HIGH( "prom.h1 ",  0x0000, 0x0100, CRC(2f98ddf0) SHA1(e90c3cebe3e788cbf8e23030f58a1153564207e2) ) // On tiles PCB
	ROM_LOAD_NIB_LOW( "prom.f1 ",   0x0000, 0x0100, CRC(adea1297) SHA1(8f365cc15cc3c26b388ba957d7cf3752584d5475) ) // On tiles PCB
	ROM_LOAD( "prom1.f1",           0x0200, 0x0020, CRC(a1130007) SHA1(9deb0eed75dd06e86f83c819a3393158be7c9dce) ) // On sprites PCB
	ROM_LOAD( "prom2.h2",           0x0220, 0x0100, CRC(76062638) SHA1(7378a26cf455d9d3df90929dc665870514c34b54) ) // On sprites PCB
That said, they have the exact same content.

**Comment by cuavas** [comment]
> That said, they have the exact same content.

So the question is, do we keep a clone for this PROM arrangement?  It may be useful for people repairing systems if they encounter this variant.

**Comment by clawgrip** [comment]
> FWIW you could use a BIOS option to load one arrangement of PROMs or the other and avoid the need for a clone.

Sounds good, will do that. Thx!!

**Review by cuavas — COMMENTED** [review]
Should the alternate PROM arrangement be added to all the sets, or only to the ones it’s been seen on in the wild?

**Comment by clawgrip** [comment]
> Should the alternate PROM arrangement be added to all the sets, or only to the ones it’s been seen on in the wild?

I've added it only to the sets I've found evidences of using both arrangements. Since it can be clearly seen on the PCB, it's easy to tell by looking at PCB pics.

Thanks

**Comment by cuavas** [comment]
Have both bootleg versions really been seen with both PROM arrangements?  That would be pretty surprising.

**Comment by clawgrip** [comment]
> Have both bootleg versions really been seen with both PROM arrangements? That would be pretty surprising.

Ah! You’re right, I’ve only seen this arrangement on original Irem/Tecfri/Williams.

Should it be removed from bootleg sets?

thx

**Comment by cuavas** [comment]
> > Have both bootleg versions really been seen with both PROM arrangements? That would be pretty surprising.
> 
> Ah! You’re right, I’ve only seen this arrangement on original Irem/Tecfri/Williams.
> 
> Should it be removed from bootleg sets?
> 
> thx

Yes, I wouldn’t add the option to the bootleg sets if they haven’t been seen with both arrangements.

**Comment by clawgrip** [comment]
> Yes, I wouldn’t add the option to the bootleg sets if they haven’t been seen with both arrangements.

Done, thanks!!

File: src/mame/irem/travrusa.cpp [hunk 1]
```diff
@@ -787,9 +800,9 @@ ROM_START( travrusab2 ) // all ROMs match travrusa but the ones where differentl
 	ROM_LOAD( "10.3k3", 0x4000, 0x2000, CRC(6fcc9fdb) SHA1(88f878b9ebf07c5a16f8cb742016cac971ed3f10) )
 
 	ROM_REGION( 0x0320, "proms", 0 )
-	ROM_LOAD( "6349-2.1k2",    0x0000, 0x0200, CRC(c9724350) SHA1(1fac20cdc0a53d94e8f67b49d7dd71d1b9f1f7ef) ) // character palette - last $100 are unused
-	ROM_LOAD( "tbp18s030.3f1", 0x0200, 0x0020, CRC(a1130007) SHA1(9deb0eed75dd06e86f83c819a3393158be7c9dce) ) // sprite palette
-	ROM_LOAD( "mb7052.3h2",    0x0220, 0x0100, CRC(76062638) SHA1(7378a26cf455d9d3df90929dc665870514c34b54) ) // sprite lookup table
+	ROM_LOAD( "mmi6349.ij",   0x0000, 0x0200, CRC(c9724350) SHA1(1fac20cdc0a53d94e8f67b49d7dd71d1b9f1f7ef) ) // character palette - last $100 are unused
+	ROM_LOAD( "tbp18s.2",     0x0200, 0x0020, CRC(a1130007) SHA1(9deb0eed75dd06e86f83c819a3393158be7c9dce) ) // sprite palette
+	ROM_LOAD( "tbp24s10.3",   0x0220, 0x0100, CRC(76062638) SHA1(7378a26cf455d9d3df90929dc665870514c34b54) ) // sprite lookup table
```
Comment (cuavas) [inline src/mame/irem/travrusa.cpp:805]: Why the change to the labels here?  This set previously had different labels to the parent and the other bootleg.  Were they found to be incorrect?

**Review by cuavas — CHANGES_REQUESTED** [review]
Please look at the overall diff.  You’ve “restored” the PROM labels on the wrong bootleg.  Now `travrusab` and `travrusab2` have their PROM labels swapped with each other.

File: src/mame/irem/travrusa.cpp (hunk 1 above)
Comment (clawgrip) [reply src/mame/irem/travrusa.cpp]: My bad. Restored the original labels. Sorry.

**Comment by clawgrip** [comment]
> Please look at the overall diff. You’ve “restored” the PROM labels on the wrong bootleg. Now `travrusab` and `travrusab2` have their PROM labels swapped with each other.

Ah! Yes, comitted before finished editing by mistake, but I saw the error, it's OK now

**Comment by cuavas** [comment]
In the future, please check your changes carefully.  It’s far too easy for errors to get in like this, and noticing/fixing them after the fact is always far more difficult.

**Comment by clawgrip** [comment]
> In the future, please check your changes carefully. It’s far too easy for errors to get in like this, and noticing/fixing them after the fact is always far more difficult.

Will do. Thanks a lot!!!!
